### PR TITLE
feat: hashing example

### PR DESCRIPTION
**Pull request by lino-levan**
This is probably an important example to have. I wish this existed as a resource when I was trying to optimize file hashing over at tea.

### CODE REVIEW

File: data/hashing.ts [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+ * @title Hashing
+ * @difficulty intermediate
+ * @tags cli, deploy
+ * @resource {https://deno.land/api?s=SubtleCrypto} Doc: crypto.subtle
+ * @resource {https://developer.mozilla.org/en-US/docs/Web/API/SubtleCrypto/digest} MDN: Cryptographic Digests
+ *
+ * Hashing data is a common operation that is facilitated
+ * through Deno's support of the Web Crypto API. There are
+ * some more advanced uses that may require extensions to
+ * the api that are provided the by the standard library.
```
Comment (kt3k) [inline data/hashing.ts:11]: typo

```suggestion
 * the api that are provided by the standard library.
```

File: data/hashing.ts [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+// Hashing a file is quite the common operation and doing this
+// without loading the whole file into memory is a common
+// requirement.
+const file = await Deno.open("example.txt", { read: true });
+
+// We can obtain an async iterable using the readable property
+const readableStream = file.readable;
+
+// We can then use this as an async iterable and has the file
```
Comment (kt3k) [inline data/hashing.ts:49]: typo?
```suggestion
// We can then use this as an async iterable and hash the file
```

**Comment by lino-levan** [comment]
I should stop writing these examples when I'm tired 😅. Thanks for catching these typos!

**Review by kt3k — APPROVED** [review]
LGTM. Thanks, Lino!

**Pull request merged**